core/src/main/scala/kafka/coordinator/group/CoordinatorLoaderImpl.scala

```diff
@@ -153,6 +154,15 @@ class CoordinatorLoaderImpl[T](
               }
 
               currentOffset = batch.nextOffset
+              val currentHighWatermark = log.highWatermark
+              if (currentOffset >= currentHighWatermark) {
+                coordinator.updateLastWrittenOffset(currentOffset)
+              }
+
+              if (currentHighWatermark > previousHighWatermark) {
+                coordinator.updateLastCommittedOffset(currentHighWatermark)
+                previousHighWatermark = currentHighWatermark
+              }
```

Comment:
I think that it works. However, it is worth pointing out that the last committed offset could be higher than the last written offset in the state machine until records past it are read.

Reply:
added a comment above